orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/load_data_files.py

```diff
@@ -1,12 +1,236 @@
+    input_defs=[InputDefinition("scratch_dataset_name", HcaScratchDatasetName)],
+    output_defs=[DynamicOutputDefinition(name="control_file_path", dagster_type=str)]
+)
+def diff_file_loads(context: AbstractComputeExecutionContext,
+                    scratch_dataset_name: HcaScratchDatasetName) -> Iterator[str]:
+    """
+    Determine files to load by joining against the target HCA dataset, then drop the result in our scratch bucket as a
+     "control file" that Jade will use for bulk file ingest
+    :param context: Dagster solid context
+    :param scratch_dataset_name: Name of the "scratch" dataset we are using for this pipeline run
+    :return: Yields a list of control file blob names
+    """
+    target_hca_dataset = context.resources.target_hca_dataset
+    file_load_table_name = 'file_load_requests'
+    bigquery_client = context.resources.bigquery_client
+    scratch = context.resources.scratch_config
+
+    _determine_files_to_load(
+        bigquery_client,
+        target_hca_dataset,
+        scratch_dataset_name,
+        file_load_table_name,
+        scratch
+    )
+
+    _extract_files_to_load_to_control_files(
+        bigquery_client,
+        context.resources.scratch_config,
+        scratch_dataset_name,
+        file_load_table_name
+    )
+
+    prefix = f'{scratch.scratch_prefix_name}/data-transfer-requests-deduped'
+    storage_client = context.resources.storage_client
+    for blob in storage_client.list_blobs(scratch.scratch_bucket_name, prefix=prefix):
+        yield DynamicOutput(
+            output_name='control_file_path',
+            value=blob.name,
+            mapping_key=blob.name.replace('/', '_').replace('-', '_')  # filter out bad chars for dagster dynamic output
+        )
+
+
+def _determine_files_to_load(
+        bigquery_client: bigquery.client.Client,
+        target_hca_dataset: TargetHcaDataset,
+        staging_dataset: HcaScratchDatasetName,
+        file_load_table_name: str,
+        scratch_config: ScratchConfig,
+) -> RowIterator:
+    fq_dataset_id = _fully_qualified_jade_dataset_name(target_hca_dataset.dataset_name)
+    query = f"""
+    WITH J AS (
+        SELECT target_path FROM `{target_hca_dataset.project_id}.{fq_dataset_id}.datarepo_load_history` WHERE state = 'succeeded'
+    )
+    SELECT S.source_path AS sourcePath, S.target_path as targetPath
+    FROM {file_load_table_name} S LEFT JOIN J USING (target_path)
+    WHERE J.target_path IS NULL
+    """
+
+    # configure the external table definition
+    job_config = bigquery.QueryJobConfig()
+    external_config: ExternalConfig = ExternalConfig(
+        ExternalSourceFormat.NEWLINE_DELIMITED_JSON
+    )
+    external_config.source_uris = [
+        f"{_gs_path_from_bucket_prefix(scratch_config.scratch_bucket_name, scratch_config.scratch_prefix_name)}/data-transfer-requests/*"
+    ]
+    external_config.schema = [
+        SchemaField("source_path", "STRING"),
+        SchemaField("target_path", "STRING")
+    ]
```

Comment:
I'm wondering if bits like this that configure the BigQuery job could be pulled out into a resource (perhaps a `BigQueryConfigurator`) or at least a separate class. This seems too procedural (and too hardcoded) to be part of the main body of the solid.